test/config_params/test_cofig_params.py

```diff
@@ -0,0 +1,368 @@
+import pytest
+import time
+import uuid
+import os
+import logging
+import json
+from urllib.parse import urlparse
+from ..common import request_start_logging,  \
+    check_events_from_splunk, request_stop_logging, \
+    start_log_producer_from_input
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    (None, 1),
+    ("history", 1)
+])
+def test_splunk_index(setup, test_input, expected):
+    '''
+    Test that user specified index can successfully index the
+    log stream from docker. If no index is specified, default
+    index "main" will be used.
+
+    Note that the HEC token on splunk side needs to be configured
+    to accept the specified index.
+    '''
+    logging.getLogger().info("testing test_splunk_index input={0} \
+                expected={1} event(s)".format(test_input, expected))
+    u_id = str(uuid.uuid4())
+
+    file_path = setup["fifo_path"]
+    start_log_producer_from_input(file_path, [("test index", False)], u_id)
+
+    index = test_input if test_input else "main"
+    request_start_logging(file_path,
+                          setup["splunk_hec_url"],
+                          setup["splunk_hec_token"],
+                          options={"splunk-index": index})
+
+    # wait for 10 seconds to allow messages to be sent
+    time.sleep(10)
+    request_stop_logging(file_path)
+
+    # check that events get to splunk
+    events = check_events_from_splunk(index=index,
+                                      id=u_id,
+                                      start_time="-1m@m",
+                                      url=setup["splunkd_url"],
+                                      user=setup["splunk_user"],
+                                      password=setup["splunk_password"])
+    logging.getLogger().info("Splunk received %s events in the last minute" +
+                             " with u_id=%s",
+                             len(events), u_id)
+    assert len(events) == expected
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    (None, 1),
```

Comment:
I believe this depends on how the application is writing logs to stdout. Do you see source being set to http::docker for all logs generated by the container? Also, whatever the working is, I feel like it should be documented.
Let me try running the tests.

Reply:
have you see another case that is not http::docker? I don't see it being set in our plugin code. I'll check docker engine and see if this is the default.

Reply:
I think you are right. From reading the docker source code: https://github.com/moby/moby/blob/f653485e57a36518085868dcb48b4d924a520877/daemon/logger/copier.go#L96
there isn't a default.  But for this test setup, http::docker is the default. I'll update the comment.

Reply:
Yes. For instance, it can be stdout or stderr. The container I created for testing I was sending logs to stderr and that's the source in Splunk.

Reply:
I updated the comment for this test. I think for this test itself, http::event is the default. The test is to test that:
- if no source is specified, default source from docker is used
- if source is specified, it is used to be the source

Reply:
ok I see the issue. http::docker is set by HEC that's tie to the splunk setup. I'll fix it